File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float n;
    int x;
    int y = 0;

    //prompt user for amount of change
    do
    {
        n = get_float("Amount of change: ");
    }

    while (n < 0);

    //convert float(dollars) into int(cents)
    x = round(n * 100);

    while (x >= 25)
    {
        x -= 25;
        y++;
    }

    while (x >= 10)
    {
        x -= 10;
        y++;
    }

    while (x >= 5)
    {
        x -= 5;
        y++;
    }

    while (x >= 1)
    {
        x -= 1;
        y++;
    }

    printf("change owed %i \n", y);

}

File: pset6/cash/cash.py
import sys

# prompt user for amount of change
while True:
    try:

        n = float(input("Amount of change: "))
    except ValueError:
        print("Enter positive number!")
        continue
    else:
        if (n < 0) == False:
            break

# convert float(dollars into int(cents)
x = round(n * 100)

coins = 0

while True:
    if (x >= 25):
        x -= 25
        coins += 1

    elif (x >= 10):
        x -= 10
        coins += 1

    elif (x >= 5):
        x -= 5
        coins += 1

    elif (x >= 1):
        x -= 1
        coins += 1

    else:
        (x == 0)
        break

# print out number of coins
print("change owed", coins)